elitzur-avro/src/main/scala/com/spotify/elitzur/converters/avro/AvroElitzurConversionUtils.scala

```diff
@@ -58,6 +58,16 @@ object AvroElitzurConversionUtils {
         schema.getTypes.asScala.map(_.getType).contains(Schema.Type.RECORD))
 
 
+  private[elitzur] def getNestedRecordSchema(schema: Schema): Schema =
```

Comment:
This seems like it's pulling a Record type from a Union type.  

Can you give context why that is needed in the `toAvro` method?

Reply:
Are you asking for a comment in the code, or an explanation in Github comments?

Here's the long-ish explanation:
This is needed because `DerivedConverter` `combine` https://github.com/spotify/elitzur/pull/5/files#diff-a884ebf83545670dc3b7b3dff638e870R328 calls `toAvro` with the field schema for each field. For fields which are optional records, we un-nest the option and call `toAvro` passing in that field schema here: https://github.com/spotify/elitzur/pull/5/files#diff-a884ebf83545670dc3b7b3dff638e870R134 . For primitive types we ignore the schema (https://github.com/spotify/elitzur/pull/5/files#diff-a884ebf83545670dc3b7b3dff638e870R61) but for record types we were taking the schema as is. This works fine if it's a required record, but if it's a union of record and null, as written pre-PR, it will fail at runtime with the following exception:
```
org.apache.avro.AvroRuntimeException: Not a record: ["null",{"type":"record","name": ...
```